task/__init__.py

```diff
@@ -312,7 +322,7 @@ def branch(context, message, pathspec=".", issue=None, push=True, branch=None, d
     return branch
 
 
-def pull(branch, body=None, issue=None, base=None, labels=['bot'], run_tests=True, dry: bool = False, **kwargs):
+def pull(branch, body=None, issue=None, base=None, labels=('bot',), run_tests=True, dry=False, **kwargs):
```

Comment:
Here as well -- this is really conceptually a list, not a tuple/record. REST/JSON don't even have a concept of a tuple. (Also, it's bad practice to have non-scalar default values anyway, these should be `None` and set inside the function)

Reply:
The bad practice is not about non-scalar value — it's about non-immutable values, and indeed, you've hit upon exactly the reason it's better to write these as `()` instead of `[]` (and this is exactly the thing that ruff complains about).

Whether its better to use an immutable non-scalar type or something like `None` is more of a matter of opinion, but I prefer this style, since it's more succinct and we don't have to have an extra check for it later.

It conceptually checks too — the default is that we want a bots label.

Incidentally, there's another way: if you type the paramter with an immutable type (like `Sequence`) then even if you use a list, ruff will consider that the list is sufficiently protected against modification.  That would require annotating this function, though, which I'm trying to avoid doing at present...

All around, I prefer to geh auf Nummer sicher and use the `()`.